libs/domain/picking/offline/data-plugin.ts

```diff
@@ -0,0 +1,106 @@
+import { OauthService } from '@spryker-oryx/auth';
+import {
+  AppEnvironment,
+  ExecPlugin,
+  InjectionPlugin,
+} from '@spryker-oryx/core';
+import { Injector } from '@spryker-oryx/di';
+import { DexieIndexedDbService } from '@spryker-oryx/indexed-db';
+import {
+  combineLatest,
+  map,
+  Observable,
+  Subscription,
+  switchMap,
+  withLatestFrom,
+} from 'rxjs';
+import { PickingListEntity, PickingProductEntity } from './entities';
+import { PickingListOnlineAdapter } from './services';
+
+export class OfflineDataPlugin extends ExecPlugin {
+  protected subscription?: Subscription;
+
+  constructor() {
+    super((app) => {
+      const injector = app!.requirePlugin(InjectionPlugin).getInjector();
+
+      const env = injector.inject(AppEnvironment, {} as AppEnvironment);
+
+      const authService = injector.inject(OauthService);
+
+      this.subscription = authService
+        .isAuthenticated()
+        .pipe(
+          switchMap((authenticated) => {
+            if (authenticated) {
+              return this.clearDb(injector).pipe(
+                switchMap(() => this.populateDb(injector))
+              );
+            }
+            return this.clearDb(injector);
```

Comment:
We should not clear DB after a logout as in the future we will keep updating the DB state in the background using push notifications.